### DISCUSSION TITLE
Bode plot?

### DISCUSSION THREAD

**Discussion by emi-netlist** [OpenHantek/OpenHantek6022#344] [Ideas]
Is there a way it can turn into a feature of openhantek or is it already and I am missing it? 
A Bode plot function up to 20 MHz would be beautiful.

**Comment by Ho-Ro** [comment]
**Impossible** - 20 Mhz cannot be measured with the Hantek hardware, you would need a sample rate > 40 MS/s.
**Please describe what you want to measure - do you have a specific problem?**
For high frequencies (from 20 kHz up to some 100 MHz) a cheap vector network analyzer is best suited, e.g. the NanoVNA for about 50 $.
For Audio measurements a sound card with the jack tools jaaa and japa (available for debian via apt) is the much better choice.

**Reply by emi-netlist** [reply]
Yes, I think you're right. I was searching for something and I've seen Hantek oscilloscopes with a function generator, but for some reason they dont offer the Bode function.

**Comment by chumariesco** [comment]
I've been also searching for a Bode plot option for audio frequencies only, mostly to measure linearity of audio devices using an external sweep generator.
The spectrum analyzer almost does this, if you follow the peak frequency while a sweep plays on the probe. But peak data is not recorded and difficult to capture with the eyes. Activating the fade option and increasing 'Digital phosphor depth' in Settings comes also very close to a bode plot, but the max. depth is '99' which amounts only for a few seconds sweep. If depth could be increased to 1000, or there would be an option for the spectrum history not to fade, it would be of great help for me.

**Reply by Ho-Ro** [reply]
> If depth could be increased to 1000

This would require much more memory with all negative drawbacks.

Instead of using an external sweep generator you could feed (bandlimited) white noise into your DUT and measure either the spectrum

![image](https://user-images.githubusercontent.com/12542359/231433862-06103d26-b41c-4d31-92ec-1d59b42c51aa.png)

or better use a dedicated audio spectrum analyser like the already proposed 
- jaaa (JACK and ALSA Audio Analyser, is an audio signal generator and spectrum analyser designed to make accurate measurements.)
- japa (Japa (JACK and ALSA Perceptual Analyser), is a 'perceptual' or 'psychoacoustic' audio spectrum analyser.
In contrast to JAAA, this is more an acoustical or musical tool than a purely technical one. Possible uses include spectrum monitoring while mixing or mastering, evaluation of ambient noise, and (using pink noise), equalisation of PA systems.

![image](https://user-images.githubusercontent.com/12542359/231432741-65386cae-84c0-488f-a37a-9a83879dabfc.png)

Fons Adriaensen has a lot of linux [audio tools](https://kokkinizita.linuxaudio.org/linuxaudio/index.html) for professional analysis available.

**Reply by chumariesco** [reply]
Much appreciated your prompt and clear response Martin. Using white noise instead didn't occur to me and it will be very useful for my future use of the scope. I will also explore those jack tools right away.
My only difficulty now with the use of the spectrum analyzer is when setting the spectrum grid under 10dB/div. As shown in the screenshot here, SP1 is at the maximum top of the screen and setting the grid to 6dB or 3dB would make the spectrum disappear. Is there a way I'm missing to move the spectrum further up, or any workaround for this?

Thanks again for the useful tips.

![Whitenoise_I2S_24xNOS](https://user-images.githubusercontent.com/60000411/231456705-4eeb2e4e-970a-4aa5-94f5-72eac5ea02e9.png)

**Reply by Ho-Ro** [reply]
You can set the reference level in menu Oscilloscope/Settings/Analysis in the top right dBV field, range is -100dB..+100dB.

![image](https://user-images.githubusercontent.com/12542359/231679180-ae4c9636-c776-4152-9d87-1d6bf534e059.png)

**Reply by chumariesco** [reply]
Perfect, thank you!

**Comment by davidswelt** [comment]
+1 to the function. I made some plots via REW, but that essentially requires going through speakers and my laptop mic (no line in available). I'd rather use the Hantek as AD converter. It would be so easy to extend the spectrogram to detect and record f0... 

(The more modular alternative would be to somehow make the incoming signal available as audio (via a virtual audio device), and then one would run REW to do more advanced audio analysis)

**Comment by sjuliancarrizo** [comment]
Hi! I came here thinking about using my 6022BE as a bode plotter, for testing passive filters up to some MHz (say 5 MHz?), by using an external function generator.
I'm thinking of something like the OpenHantek software storing Vpp values of each channel (say CH1 as input and CH2 as output), for each frequency change the function generator does. Then that data can be exported as CSV, and bode plots can be easily obtained with Calc or Excel.
Of course that doing all of that in the same OpenHantek GUI would be amazing! But at least being able to export the data, would help a lot when trying to obtain Bode plots from a filter.

I would be pleased to contribute on this functionality...

Any thoughts on this?

**Reply by jonw0224** [reply]
My thoughts on the easiest way to implement this. Add a feature to save the measured values to a CSV like a data recorder. You could have the option to save at regular intervals or on each waveform refresh. Each row would contain a timestamp and then each measurement value. You'd set it up to save the Vpp values of Ch1 and Ch2 as well as the frequency values of Ch1 and Ch2. It would be really cool if the feature would save the waveform sample information too.

**Reply by jiangge12** [reply]
Ho-Ro's another project Hantek6022API provide all the hardware control , but I met some USB related errors while try it's samples. 

lucky found another project  https://github.com/BMecke/hantekosc , also base on Hantek6022API , and works.

then you will get data sets from Hantek6022 after few python coding.    [ add digital filter:  https://jiangge12.github.io/Pic_DSO.png ]

as to plot, filter ,FFT, save .... all yours

**Reply by jonw0224** [reply]
Interesting. I'll try that out!
